pkg/components/render_test.go

```diff
@@ -0,0 +1,73 @@
+package components
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/openshift/microshift/pkg/assets"
+	"github.com/openshift/microshift/pkg/config/lvmd"
+)
+
+func sourceTemplate() []byte {
+	b, err := assets.Asset("assets/components/odf-lvm/topolvm-lvmd-config_configmap_v1.yaml")
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+func Test_renderLvmd(t *testing.T) {
+	var lvmdCM = []byte(
+		`# Source: topolvm/templates/lvmd/configmap.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: lvmd
+  namespace: openshift-storage
+data:
+  lvmd.yaml: "device-classes:\n- name: test_class\n  volume-group: test_volume_group\n  default: true\n  spare-gb: 2\nsocket-name: /run/lvmd/lvmd.socket\n"
+`)
+	var dc = []*lvmd.DeviceClass{
+		{
+			Name:        "test_class",
+			VolumeGroup: "test_volume_group",
+			Default:     true,
+			SpareGB:     2,
+		},
+	}
+
+	type args struct {
+		b []byte
+		p assets.RenderParams
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "",
+			args: args{
+				b: sourceTemplate(),
+				p: map[string]string{
+					"lvmd": lvmd.NewLVMDConfig(dc).String(),
+				},
+			},
+			want:    lvmdCM,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := renderLvmd(tt.args.b, tt.args.p)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("renderLvmd() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !bytes.Equal(got, tt.want) {
```

Comment:
https://github.com/stretchr/testify has some nice features for showing where differences are in string comparisons like this. In another PR we should look at what adding that would do to the dependency list.

Reply:
I like testify, but yeah, I didn't want to stack on more dependencies for this PR.